File: Project-Euler/JavaScript/Problem3.js
// What is the largest prime factor of the given number?

function checkPrime(n) {
    for (let i = 2; i < n; i++)
        if (n % i === 0) return false;
    return true;
}
function largestPrimeFactor(number) {
    let ans = 1;
    for (let i = 1; i <= number / ans; i++) if (number % i == 0 && checkPrime(i) && ans < i) ans = i;
    return ans;
}

console.log(largestPrimeFactor(13195));

File: Project-Euler/C++/Problem3.cpp
// What is the largest prime factor of a given number N ?

#include <iostream>
#include <math.h>
using namespace std;

int main(){
    int t;
    cin >> t;
    for(int a0 = 0; a0 < t; a0++){
        long long n;
        cin >> n;
        while(n%2 == 0)
            n /= 2;
        if(n == 1)
            cout << 2 << "\n";
        else {
            for(long int i = 3; i <= sqrt(n); i += 2) {
                while (n%i == 0)
                    n /= i;
                if(n == 1)
                    cout << i << "\n";
            }
            if(n != 1)
                cout << n << "\n";
        }
    }
    return 0;
}
